text/0045-add-notion-of-protocol.md

```diff
@@ -0,0 +1,144 @@
+# Summary
+- Define supported protocols for services and outputs in libobs and OBS Studio
+- Outputs for streaming are registered with one or more compatible protocols and codecs
+- Only codecs compatible with the output and the service are shown
+- Audio codec can be chosen if various compatible, codec in simple mode and encoder advanced output mode.
+- Will be extended with RFC 39
+
+# Motivation
+Create a better management of outputs, encoders, and services with an approach focused on protocols.
+
+# Design
+
+## Outputs API
+Only outputs with the `OBS_OUTPUT_SERVICE` flag are concerned.
+
+Adding to `obs_output_info`:
+- `const char *protocols`: protocols separated with a semi-colon (ex: `"RTMP;RTMPS"`), required to register the output.
+- `const char *protocols_prefixes`: URL protocols prefixes separated with a semi-colon `"rtmp://;rtmps://"`, optional.
+  - Up to one URL prefix per protocol in `protocols`.
+  - Those URL prefixes must be in the same order as `protocols`.
+  - Each URL prefix must be unique to the protocol
+  - Meant to be used for protocol auto-detection on server URL.
```

Comment:
Is auto-detection used for services in services.json, or is auto-detection _only_ used for user-specified Custom Server URLs?

Reply:
Both.